pallets/avn-anchor/src/tests.rs

```diff
@@ -0,0 +1,626 @@
+use crate::{
+    mock::*, tests::RuntimeCall, CheckpointId, Error, Event, NextCheckpointId,
+    REGISTER_CHAIN_HANDLER, SUBMIT_CHECKPOINT, UPDATE_CHAIN_HANDLER,
+};
+use codec::Encode;
+use frame_support::{assert_noop, assert_ok, BoundedVec};
+use pallet_avn_proxy::Error as avn_proxy_error;
+use sp_avn_common::Proof;
+use sp_core::{sr25519, ConstU32, Pair, H256};
+use sp_runtime::traits::Hash;
+
+fn create_account_id(seed: u8) -> AccountId {
+    create_account_pair(seed).public()
+}
+
+fn create_account_pair(seed: u8) -> sr25519::Pair {
+    sr25519::Pair::from_seed(&[seed; 32])
+}
+
+fn bounded_vec(input: &[u8]) -> BoundedVec<u8, ConstU32<32>> {
+    BoundedVec::<u8, ConstU32<32>>::try_from(input.to_vec()).unwrap()
+}
+
+fn create_proof(
+    signer_pair: &sr25519::Pair,
+    relayer: &AccountId,
+    payload: &[u8],
+) -> Proof<Signature, AccountId> {
+    let signature = Signature::from(signer_pair.sign(payload));
+    Proof { signer: signer_pair.public(), relayer: relayer.clone(), signature }
+}
+
+#[test]
+fn register_chain_handler_works() {
+    new_test_ext().execute_with(|| {
+        let handler = create_account_id(1);
+        let name = bounded_vec(b"Test Chain");
+
+        assert_ok!(AvnAnchor::register_chain_handler(RuntimeOrigin::signed(handler), name.clone()));
+
+        let chain_data = AvnAnchor::chain_handlers(handler).unwrap();
+        assert_eq!(chain_data, 0);
+        let chain_data = AvnAnchor::chain_data(0).unwrap();
+        assert_eq!(chain_data.name, name);
+        assert_eq!(AvnAnchor::nonces(chain_data.chain_id), 0);
+
+        System::assert_last_event(Event::ChainHandlerRegistered(handler, 0, name).into());
+    });
+}
+
+#[test]
+fn register_chain_handler_fails_for_existing_handler() {
+    new_test_ext().execute_with(|| {
+        let handler = create_account_id(1);
+        let name = bounded_vec(b"Test Chain");
+
+        assert_ok!(AvnAnchor::register_chain_handler(RuntimeOrigin::signed(handler), name.clone()));
+
+        assert_noop!(
+            AvnAnchor::register_chain_handler(
+                RuntimeOrigin::signed(handler),
+                bounded_vec(b"Another Chain")
+            ),
+            Error::<TestRuntime>::HandlerAlreadyRegistered
+        );
+    });
+}
+
+#[test]
+fn register_chain_handler_fails_for_empty_name() {
+    new_test_ext().execute_with(|| {
+        let handler = create_account_id(1);
+        let empty_name = bounded_vec(b"");
+
+        assert_noop!(
+            AvnAnchor::register_chain_handler(RuntimeOrigin::signed(handler), empty_name),
+            Error::<TestRuntime>::EmptyChainName
+        );
+    });
+}
+
+// #[test]
```

Comment:
I think it should be ok to have multiple chain names because the handler and chainId will be different so we can remove this test. But I would prefer to merge as is and then do a separate PR to remove it.

Reply:
why check in, commented out code?